plugins/services/src/js/components/ServiceConfigDisplay.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+
+import ConfigurationMap from '../../../../../src/js/components/ConfigurationMap';
+import ConfigurationMapHeading from '../../../../../src/js/components/ConfigurationMapHeading';
+import ConfigurationMapLabel from '../../../../../src/js/components/ConfigurationMapLabel';
+import ConfigurationMapRow from '../../../../../src/js/components/ConfigurationMapRow';
+import ConfigurationMapSection from '../../../../../src/js/components/ConfigurationMapSection';
+import ConfigurationMapValue from '../../../../../src/js/components/ConfigurationMapValue';
+import ServiceConfigDisplayUtil from '../utils/ServiceConfigDisplayUtil';
+import Util from '../../../../../src/js/utils/Util';
+
+const shouldExcludeItem = (row, appDefinition) => {
+  switch (row.key) {
+    case 'gpus':
+      return appDefinition.gpus === 0;
+    case 'container.volumes':
+      return appDefinition.container == null
+        || appDefinition.container.volumes == null
+        || appDefinition.container.volumes.length === 0;
+    case 'healthChecks':
+      return appDefinition.healthChecks == null
+        || appDefinition.healthChecks.length === 0;
+    case 'secrets':
+      return appDefinition.secrets == null
+        || Object.keys(appDefinition.secrets).length === 0;
+    case 'env':
+      return appDefinition.env == null
+        || Object.keys(appDefinition.env).length === 0;
+    case 'labels':
+      return appDefinition.labels == null
+        || Object.keys(appDefinition.labels).length === 0;
+    default:
+      return false;
+  }
+};
+
+const ServiceConfigDisplay = ({appDefinition}) => {
+  const configurationMapSections = ServiceConfigDisplayUtil.displayValues.map(
+    (section, sectionIndex) => {
+      let configurationMapRows = section.values.reduce(
```

Comment:
Instead of reduce, you could do a `filter` and then `map`, which I think is a little bit more readable, and makes the function easier to parse. Refactored it a bit here (didn't test it):
```js
const getDisplayValue = (type, value) => {
  // Otherwise we treat the row as "label:value" type display.
  let displayValue = null;

  // If the row's type is pre, we wrap it in a pre tag.
  if (type === 'pre') {
    return (
      <pre className="flush transparent wrap">{value}</pre>
    );
  }

  return ServiceConfigDisplayUtil.getDisplayValue(value);
}

const ServiceConfigDisplay = ({appDefinition}) => {
  const configurationMapSections = ServiceConfigDisplayUtil.displayValues.map(
    (section, sectionIndex) => {
      let configurationMapRows = section.values.filter((row) => {
          // Some rows must be excluded if relevant data is missing.
          return !shouldExcludeItem(row, appDefinition);
        })
        .map((row, rowIndex) => {
          let reactKey = `${sectionIndex}.${rowIndex}`;
          let value = Util.findNestedPropertyInObject(appDefinition, row.key);

          // If a transformValue was specified on the row, we use it.
          if (row.transformValue != null) {
            value = row.transformValue(value, appDefinition);
          }

          if (row.render != null) {
            // If a custom render method was specified on the row, we use that.
            return row.render(value, appDefinition);
          }

          if (row.heading != null) {
            // If the row is a heading, we render the heading.
            return (
              <ConfigurationMapHeading key={reactKey} level={row.headingLevel}>
                {row.heading}
              </ConfigurationMapHeading>
            );
          }

          return (
            <ConfigurationMapRow key={reactKey}>
              <ConfigurationMapLabel>
                {row.label}
              </ConfigurationMapLabel>
              <ConfigurationMapValue>
                {getDisplayValue(row.type, value)}
              </ConfigurationMapValue>
            </ConfigurationMapRow>
          );
        }
      );

      if (configurationMapRows.length === 0) {
        return null;
      }

      return (
        <ConfigurationMapSection key={sectionIndex}>
          {configurationMapRows}
        </ConfigurationMapSection>
      );
    }
  );
  ...
}
```